docs/behat/basic-usage.rst

```diff
@@ -0,0 +1,510 @@
+                filters:
+                    tags: "@managing_countries && @ui"
+
+First essential thing that happens here is configuration of tags, from now behat will be searching all your features tagged with ``@managing_countires`` and your scenarios tagged with ``@ui``.
+Second thing is ``contexts_as_services:`` in this section we will be placing all our services with step implementation.
+
+I mentioned about side effects of ubiquitous language that's why we can very easy switch our testing context to ``@domain``. Domain suite will look like:
+
+.. code-block:: yaml
+
+    default:
+        suites:
+            domain_managing_countries:
+                contexts_as_services:
+                    - sylius.behat.context.hook.doctrine_orm
+
+                    - sylius.behat.context.transform.country
+                    - sylius.behat.context.transform.shared_storage
+
+                    - sylius.behat.context.setup.geographical
+                    - sylius.behat.context.setup.security
+
+                    - sylius.behat.context.domain.admin.managing_countries // Domain step implementation.
+                filters:
+                    tags: "@managing_countries && @domain"
+
+We have almost finished with suite configuration now we need to register our first behat context service, but before we will do it we need one more thing because like we said this is ui feature so we need page objects.
+
+Registering Pages
+-----------------
+Page object pattern allows us to hide all interaction with html behind those objects so all ui things happens here.
+
+We got three kinds of page:
+    - Page // First layer of our pages it knows how to interact with dom objects. It has method ->getUrl(array $urlParameters) where you can define raw url to open it.
+    - SymfonyPage // This page extends Page and it has router injected so from now ->getUrl() generates url from route name which gets from ->getRouteName() method.
+    - Base Crud Pages (IndexPage, CreatePage, UpdatePage) // This page extends SymfonyPage and it's specific to sylius resources it has resource name injected and from now it knows about route name.
+
+There are two ways to manipulate html. By using ``->getDocument()`` or ``->getElement('your_element')`` first method will return ``DocumentElement`` which represents html structure of currently opened page,
+second one is more tricky it use ``->getDefinedElements()`` method and it will return ``NodeElement`` which represents only restricted html structure.
+
+Usage example of ``->getElement('your_element')`` and ``->getDefinedElements`` methods.
+
+.. code-block:: php
+
+    class CreatePage extends SymfonyPage implements CreatePageInterface
+    {
+        /**
+         *{@inheritdoc}
+         */
+        protected function getDefinedElements() // This method returns simple associative array, where key is name of your element and value is locator.
```

Comment:
`// This method returns a simple associative array, where the key is the name of your element and the value is its locator.`